pkg/source/kinesis/kinesis_source_test.go

```diff
@@ -339,15 +344,104 @@ func TestKinesisSourceHCL(t *testing.T) {
 			assert.NotNil(result)
 			assert.Nil(err)
 
-			if !reflect.DeepEqual(result, tt.Expected) {
-				t.Errorf("GOT:\n%s\nEXPECTED:\n%s",
-					spew.Sdump(result),
-					spew.Sdump(tt.Expected))
+			resultConf, ok := result.(*Configuration)
+			if !ok {
+				t.Fatal("result is not of type pointer to Configuration")
 			}
+
+			assert.Equal(resultConf.StreamName, tt.Expected.StreamName)
+			assert.Equal(resultConf.Region, tt.Expected.Region)
+			assert.Equal(resultConf.AppName, tt.Expected.AppName)
+			assert.Equal(resultConf.RoleARN, tt.Expected.RoleARN)
+			assert.Equal(resultConf.StartTimestamp, tt.Expected.StartTimestamp)
+			assert.Equal(resultConf.ConcurrentWrites, tt.Expected.ConcurrentWrites)
+			assert.Equal(resultConf.ReadThrottleDelayMs, tt.Expected.ReadThrottleDelayMs)
+			assert.Equal(resultConf.ShardCheckFreqSeconds, tt.Expected.ShardCheckFreqSeconds)
+			assert.Equal(resultConf.LeaderActionFreqSeconds, tt.Expected.LeaderActionFreqSeconds)
+
+			if !tt.ClientNameUUID {
+				assert.Equal(resultConf.ClientName, tt.Expected.ClientName)
+			} else {
+				_, err := uuid.Parse(resultConf.ClientName)
+				assert.Nil(err)
+			}
+
 		})
 	}
 }
 
+func TestClientNamePassThrough(t *testing.T) {
```

Comment:
What's the purpose of this additional test?

Can it be described in a comment?

Reply:
Discussed on call - the test above does all we need, we can remove this one.